### DISCUSSION TITLE
DB 복구 시에 썼던 리두로그, InnoDB 리두 로그 버퍼는 뭘까요?

### DISCUSSION THREAD

**Discussion by CODe5753** [Well-Founded-Confidence/RealMySQL#11] [Q&A]
**데이터가 변경될 때 그 기록을 저장하는 것을 리두로그(Redo Log)**라고 하는데요.
이 때 이 작업을 캐싱(로그 버퍼)하는 이유가 뭘까요?

**Comment by menthamin** [comment]
Binlog vs. Redo Log
- 공통점: 변경 사항을 기록 함
- Redo Log는 휘발

Redo Log
- Select를 제외한 변경작업을 Insert, Update, Delete 를 기록
- 트랜잭션의 특징 Atomic 보장 (전부다 실행되거나 아예 실행되지 않거나)

Binlog (바이너리 로그)
- DB를 날렸을 때 복구를 위해 사용
- 수정, 삭제, 추가의 모든 내용을 기록 함
- 별다른 복구 설정이 없을 때 이를 사용해 복구

Q. Binlog 데이터가 어마어마할텐데 백업 전략이 있을지? (나현님)
- DB 생성 시점 부터 있어야 백업에 활용 가능
- Binlog는 바로 사용하지 못하고 변환이 필요

Q2. 현업에서 DB 환경 어떻게 백업하고 있는지? (승현님)
- 서비스, 스테이지, 개발 환경의 Configure 관리
- 개발: VPN으로 접근 가능
- 스테이지(회사마다 테스트, QA): 허용된 PC를 한번 거쳐서만 접근 가능 (베스천호스트..?)
- 서비스: 접근 불가..

라이브 서비스에 적용되는 DML은 DBA 분에게 전달해서 수정의뢰를 한다.

프로시저를 이용해 업데이트를 해야하는 경우가 있었는데
프로시저가 업데이트가 되어있지 않아 이슈가 발생한적 있음.

# 개발, 스테이지, 서비스의 구분
- 회사의 조직 규모, 서비스의 중요도 등에 따라 환경 구성이 다를 수 있다.
